shopfloor_reception/services/reception.py

```diff
@@ -0,0 +1,877 @@
+                selected_lines.qty_done += packaging.qty
+                return self._response_for_set_quantity(picking, selected_lines)
+            package = search.package_from_scan(barcode)
+            dest_location = selected_lines.location_dest_id
+            child_locations = self.env["stock.location"].search(
+                [("id", "child_of", dest_location.id)]
+            )
+            if package:
+                pack_location = package.location_id
+                if pack_location:
+                    if pack_location not in child_locations:
+                        # If the scanned package has a location that isn't a child
+                        # of the move dest, return an error
+                        message = {
+                            "message_type": "error",
+                            "body": _("You cannot place it here."),
+                        }
+                        return self._response_for_set_quantity(
+                            picking, selected_lines, message=message
+                        )
+                    else:
+                        # If the scanned package has a valid destination,
+                        # set both package and destination on the package,
+                        # and go back to the selection line screen
+                        selected_lines.result_package_id = package
+                        selected_lines.location_dest_id = pack_location
+                        return self._response_for_select_line(picking)
+                # Scanned package has no location, move to the location selection
+                # screen
+                selected_lines.result_package_id = package
+                return self._response_for_set_destination(picking, selected_lines)
+            location = search.location_from_scan(barcode)
+            if location:
+                if location not in child_locations:
+                    # Scanned location isn't a child of the move's dest location
+                    message = {
+                        "message_type": "error",
+                        "body": _("You cannot place it here."),
+                    }
+                    return self._response_for_set_quantity(
+                        picking, selected_lines, message=message
+                    )
+                # process without pack, set destination location, and go back to
+                # `select_line`
+                selected_lines.location_dest_id = location
+                return self._response_for_select_line(picking)
+            if not confirmation:
+                return self._response_for_set_quantity(
+                    picking,
+                    selected_lines,
+                    message={
+                        "message_type": "warning",
+                        "body": _("Put in this new PACK {}?").format(barcode),
+                    },
+                )
+            package = self.env["stock.quant.package"].create({"name": barcode})
```

Comment:
The creation of a package wasn't optional (reading the specs)?

> Option on menu item:
> - Always put in Pack (SSCC)
>

Reply:
Well, regarding the specs, I should have put that in the shopfloor module.
I thought it would be smarter to create a new module. If the module is installed, then I suppose that this option is enabled by default.

Reply:
It has its place in this module I think as we generally create packages when receiving the goods. And the option is enabled on the menu, so we could have two different menus to receive goods, with the option enabled only on one.

Reply:
At this stage, it looks good, all the tests have been done and the pack can now be created